optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/shift/Shift.java

```diff
@@ -107,17 +137,26 @@ public long getLengthInHours() {
         return startDateTime.until(endDateTime, ChronoUnit.HOURS);
     }
 
+    public boolean isMoved() {
+        return originalEmployee != null && originalEmployee != employee;
+    }
+
+    public boolean hasRequiredSkills() {
+        return employee.getSkillProficiencySet().containsAll(spot.getRequiredSkillSet()) &&
+                employee.getSkillProficiencySet().containsAll(requiredSkillSet);
```

Comment:
This smells of inefficiency. How about:
    
    Set<...> skillProficiencySet = employe.getSkillProficiencySet();
    return Stream.concat(spot.getRequiredSkillSet().stream(), requiredSkillSet.stream())
        .distinct() // Test every skill at most once.
        .allMatch(skillProficiencySet::contains);

Assuming that there is a significant overlap between the two skill sets, this will be faster. Otherwise, not so much. Maybe not worth the hassle.

Reply:
There should be 0 overlap between the spot skill set and the shift skill set (shift skill set is additional skills required for the shift).